File: app/src/main/java/ru/mamykin/foreignbooksreader/retrofit/YandexTranslateService.java
package ru.mamykin.foreignbooksreader.retrofit;

import retrofit2.http.GET;
import retrofit2.http.Query;
import ru.mamykin.foreignbooksreader.models.Translation;
import rx.Observable;

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
public interface YandexTranslateService {
    String BASE_URL = "https://translate.yandex.net/";

    @GET("api/v1.5/tr.json/translate")
    Observable<Translation> translate(
            @Query("key") String key,
            @Query("text") String text,
            @Query("lang") String lang,
            @Query("format") String format,
            @Query("options") String options
    );
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/retrofit/YandexTranslateService.kt
package ru.mamykin.foreignbooksreader.retrofit

import retrofit2.http.GET
import retrofit2.http.Query
import ru.mamykin.foreignbooksreader.models.Translation
import rx.Observable

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
interface YandexTranslateService {

    @GET("api/v1.5/tr.json/translate")
    fun translate(
            @Query("key") key: String,
            @Query("text") text: String,
            @Query("lang") lang: String,
            @Query("format") format: String,
            @Query("options") options: String
    ): Observable<Translation>

    companion object {
        val BASE_URL = "https://translate.yandex.net/"
    }
}